stonesoup/hypothesiser/distance.py

```diff
@@ -87,3 +91,65 @@ def hypothesise(self, track, detections, timestamp, **kwargs):
                         measurement_prediction))
 
         return MultipleHypothesis(sorted(hypotheses, reverse=True))
+
+
+class MultiDistanceHypothesiser(Hypothesiser):
+    """Prediction Hypothesiser based on a set of Measures
+
+    Similar to the :class:`~.DistanceHypothesiser` this class scores prediction-detection pairs
+    using their distance in a measurement space. However, this class attempts to use the
+    detection's measurement model to determine a suitable Measure and distance to use. These are
+    provided by the user.
+    To do so, the class utilises a list of :class:`~.DistanceHypothesiser` and corresponding
+    measurement models. When a detection is received of particular model, the class will attempt
+    to find the model's corresponding hypothesiser and use its output.
+    """
+
+    hypothesisers: Sequence[DistanceHypothesiser] = Property(
+        doc="Sequence of hypothesisers to utilise output of.")
+    measurement_models: Sequence[MeasurementModel] = Property(
+        doc="Sequence of measurement models that will be compared against incoming detections. "
+            "Must be same length as hypothesisers sequence."
+    )
```

Comment:
Would these be better as `Mapping[MeasurementModel, Hypothesiser]`?

Reply:
Yes, definitely.